backend/src/test/unimplemented_resources/ID025_respondToCollaborationRequests.feature

```diff
@@ -0,0 +1,73 @@
+      | id | ideaId | userId | status   | additionalContact | message                                                                       |
+      | 23 | 17     | 2      | Pending  | "438-764-1940"    | "Hi, I would like to join this project!"                                      |
+      | 24 | 18     | 2      | Pending  | null              | "Hi, I have experience in that field and I am interested in working on this!" |
+      | 25 | 20     | 3      | Accepted | null              | "Hi, I have experience in that field and I am interested in working on this!" |
+      | 26 | 21     | 2      | Pending  | null              | "Hi, I have experience in that field and I am interested in working on this!" |
+
+    And the database contains the following collaboration confirmation:
+      | id | collaborationRequestId | additionalContact                 | message                                                          |
+      | 27 | 23                     | "Hi, my contact is +438-764-1940" | "Hi, could you develop on your experience in that field please." |
+      | 28 | 24                     | ""                                | ""                                                               |
+
+  # Normal/alternate flows
+  Scenario Outline: User successfully accepts collaboration request for their created idea
+    Given I am logged in as the user with email "<email>" and password "<password>"
+    When the user accepts the collaboration request with id "<collaboration_request_id>"
+    Then the collaboration request with id "<collaboration_request_id>" has status "Accepted"
+    Then the collaboration confirmation with id "<collaboration_confirmation_id>" is created and the message "<message>" is sent to the user with id "<user_id>"
+    And the person who sent the request receives a custom message from the user with their contact information
+
+    Examples:
+      | email                   | password     | collaboration_request_id | collaboration_confirmation_id | message                                                          | user_id |
+      | john.goblikon@gmail.com | P@ssWord1234 | 23                       | 27                            | "Hi, could you develop on your experience in that field please." | 2       |
+      | jane.doe@gmail.com      | P@ssWord1234 | 24                       | 28                            | ""                                                               | 2       |
+
+  Scenario Outline: User successfully declines collaboration request for their created idea
+    Given I am logged in as the user with email "<email>" and password "<password>"
+    When the user declines the collaboration request with id "<collaboration_request_id>"
+    Then the collaboration request with id "<collaboration_request_id>" has status "Declined"
+
+    Examples:
+      | email                    | password     | collaboration_request_id |
+      | wassim.jabbour@gmail.com | P@ssWord1234 | 26                       |
```

Comment:
Looks awesome! Why don't we add an error flow where the user tries to change the status from either Accepted or Declined to something else, it should not let them. i.e. You can only transition from the Pending status, and once it is either accepted or declined that state is final. It can return an error message like "This collaboration request has already been declined/accepted (whichever applies) and its status cannot be changed". We can also have the error flow where a user tries to change the status of a colab request that does not attach to an idea that they own